packages/api/src/providers/fitbit.ts

```diff
@@ -359,21 +372,124 @@ export class Fitbit extends Provider implements OAuth2 {
       [FitbitScopes.weight]: "body",
     };
 
-    const subscriptionId = userId;
+    const activeSubscriptions = await this.getActiveSubscriptions(accessToken);
+    if (activeSubscriptions)
+      await this.revokeTokenFromOtherUsers(user, fitbitUserId, activeSubscriptions, accessToken);
 
     // TODO #652: Implement userRevokedAccess
 
+    // Creates new WH subscriptions based on the user's selected scopes
     await Promise.all(
       Object.entries(subscriptionTypes).map(async ([key, subscriptionType]) => {
         if (scopes.includes(key)) {
-          const subscriptionUrl = `${Fitbit.URL}/${Fitbit.API_PATH}/${subscriptionType}/apiSubscriptions/${subscriptionId}-${subscriptionType}.json`;
+          const subscriptionId = `${fitbitUserId}-${subscriptionType}`;
+          const subscriptionUrl = `${Fitbit.URL}/${Fitbit.API_PATH}/${subscriptionType}/apiSubscriptions/${subscriptionId}.json`;
           this.createSubscription(subscriptionUrl, accessToken);
         }
       })
     );
   }
 
-  // Creates a subscription for all or one specific collectionType
+  // Finds existing users connected to this Fitbit account, revokes their tokens and sends WH updates about the disconnections
+  async revokeTokenFromOtherUsers(
+    currentUser: ConnectedUser,
+    fitbitUserId: string,
+    activeSubscriptions: FitbitWebhookSubscriptions,
+    accessToken: string
+  ): Promise<void> {
+    try {
+      if (activeSubscriptions.length > 0) {
+        const connectedUsers = await getConnectedUsersByTokenOrFail(
+          ProviderSource.fitbit,
+          fitbitUserId
+        );
+
+        await Promise.all(
+          connectedUsers.map(async user => {
+            if (
+              user.dataValues.id !== currentUser.dataValues.id &&
+              user.dataValues.cxId === currentUser.dataValues.cxId
+            ) {
+              await this.oauth.revokeLocal(user);
+              user = await getConnectedUserOrFail({
+                id: user.id,
+                cxId: user.cxId,
+              });
+              sendProviderDisconnected(user, [ProviderSource.fitbit]);
+            }
+          })
+        );
+        await this.deleteSubscriptions(activeSubscriptions, accessToken);
+      }
+    } catch (err) {
+      console.log(
+        `Failed to revoke the token of a Fitbit user. User: ${currentUser}, Error: ${err}`
+      );
+      capture.error(err, {
+        extra: { context: "fitbit.revokeTokenFromExistingUsers", err, user: currentUser },
+      });
```

Comment:
General comment - why are we always catching errors but never re-throwing them?

Do we really want to continue the code execution if there is a failure in one of the funcs?

Reply:
Added re-throwing of errors

Reply:
I don't see the error being rethrown here?